static/src/javascripts/projects/common/modules/navigation/user-account.ts

```diff
@@ -0,0 +1,141 @@
+import { getUserFromCookie, isUserLoggedIn } from 'common/modules/identity/api';
+import fastdom from 'lib/fastdom-promise';
+import { bufferedNotificationListener } from '../bufferedNotificationListener';
+
+const updateCommentLink = (commentItems: Element[]): void => {
+	const user = getUserFromCookie();
+
+	if (user) {
+		commentItems.forEach((commentItem) => {
+			void fastdom
+				.measure(() =>
+					commentItem.querySelector('.js-add-comment-activity-link'),
+				)
+				.then((commentLink) => {
+					if (commentLink) {
+						void fastdom.mutate(() => {
+							commentItem.classList.remove('u-h');
+							commentLink.setAttribute(
+								'href',
+								`https://profile.theguardian.com/user/id/${user.id}`,
+							);
+						});
+					}
+				});
+		});
+	}
+};
+
+const groupNotificationsByTarget = (
+	notifications: NotificationEvent[],
+): Record<string, string[]> => {
+	const notificationsMap: Record<string, string[]> = {};
+	notifications.forEach(({ message, target }) => {
+		if (Array.isArray(notificationsMap[target])) {
+			notificationsMap[target].push(message);
+		} else {
+			notificationsMap[target] = [message];
+		}
+	});
+	return notificationsMap;
+};
```

Comment:
Just an FYI and not a blocker, but I think this might be slightly more explicit if using a reducer.

Reply:
Or if you’re feeling bold and we’re transpiling it: [`group`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/group)!